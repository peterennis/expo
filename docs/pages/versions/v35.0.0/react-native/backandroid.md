---
id: backandroid
title: BackAndroid
---

Deprecated. Use BackHandler instead.

### Methods

- [`exitApp`](../backandroid/#exitapp)
- [`addEventListener`](../backandroid/#addeventlistener)
- [`removeEventListener`](../backandroid/#removeeventlistener)

---

# Reference

## Methods

### `exitApp()`

```javascript

static exitApp()

```

---

### `addEventListener()`

```javascript

static addEventListener(eventName, handler)

```

---

### `removeEventListener()`

```javascript

static removeEventListener(eventName, handler)

```
