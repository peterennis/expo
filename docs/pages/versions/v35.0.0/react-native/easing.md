---
id: easing
title: Easing
---

The `Easing` module implements common easing functions. This module is used by [Animated.timing()](../animated/#timing) to convey physically believable motion in animations.

You can find a visualization of some common easing functions at http://easings.net/

### Predefined animations

The `Easing` module provides several predefined animations through the following methods:

- [`back`](../easing/#back) provides a simple animation where the object goes slightly back before moving forward
- [`bounce`](../easing/#bounce) provides a bouncing animation
- [`ease`](../easing/#ease) provides a simple inertial animation
- [`elastic`](../easing/#elastic) provides a simple spring interaction

### Standard functions

Three standard easing functions are provided:

- [`linear`](../easing/#linear)
- [`quad`](../easing/#quad)
- [`cubic`](../easing/#cubic)

The [`poly`](../easing/#poly) function can be used to implement quartic, quintic, and other higher power functions.

### Additional functions

Additional mathematical functions are provided by the following methods:

- [`bezier`](../easing/#bezier) provides a cubic bezier curve
- [`circle`](../easing/#circle) provides a circular function
- [`sin`](../easing/#sin) provides a sinusoidal function
- [`exp`](../easing/#exp) provides an exponential function

The following helpers are used to modify other easing functions.

- [`in`](../easing/#in) runs an easing function forwards
- [`inOut`](../easing/#inout) makes any easing function symmetrical
- [`out`](../easing/#out) runs an easing function backwards

### Methods

- [`step0`](../easing/#step0)
- [`step1`](../easing/#step1)
- [`linear`](../easing/#linear)
- [`ease`](../easing/#ease)
- [`quad`](../easing/#quad)
- [`cubic`](../easing/#cubic)
- [`poly`](../easing/#poly)
- [`sin`](../easing/#sin)
- [`circle`](../easing/#circle)
- [`exp`](../easing/#exp)
- [`elastic`](../easing/#elastic)
- [`back`](../easing/#back)
- [`bounce`](../easing/#bounce)
- [`bezier`](../easing/#bezier)
- [`in`](../easing/#in)
- [`out`](../easing/#out)
- [`inOut`](../easing/#inout)

---

# Reference

## Methods

### `step0()`

```javascript

static step0(n)

```

A stepping function, returns 1 for any positive value of `n`.

---

### `step1()`

```javascript

static step1(n)

```

A stepping function, returns 1 if `n` is greater than or equal to 1.

---

### `linear()`

```javascript

static linear(t)

```

A linear function, `f(t) = t`. Position correlates to elapsed time one to one.

http://cubic-bezier.com/#0,0,1,1

---

### `ease()`

```javascript

static ease(t)

```

A simple inertial interaction, similar to an object slowly accelerating to speed.

http://cubic-bezier.com/#.42,0,1,1

---

### `quad()`

```javascript

static quad(t)

```

A quadratic function, `f(t) = t * t`. Position equals the square of elapsed time.

http://easings.net/#easeInQuad

---

### `cubic()`

```javascript

static cubic(t)

```

A cubic function, `f(t) = t * t * t`. Position equals the cube of elapsed time.

http://easings.net/#easeInCubic

---

### `poly()`

```javascript

static poly(n)

```

A power function. Position is equal to the Nth power of elapsed time.

n = 4: http://easings.net/#easeInQuart n = 5: http://easings.net/#easeInQuint

---

### `sin()`

```javascript

static sin(t)

```

A sinusoidal function.

http://easings.net/#easeInSine

---

### `circle()`

```javascript

static circle(t)

```

A circular function.

http://easings.net/#easeInCirc

---

### `exp()`

```javascript

static exp(t)

```

An exponential function.

http://easings.net/#easeInExpo

---

### `elastic()`

```javascript

static elastic(bounciness)

```

A simple elastic interaction, similar to a spring oscillating back and forth.

Default bounciness is 1, which overshoots a little bit once. 0 bounciness doesn't overshoot at all, and bounciness of N > 1 will overshoot about N times.

http://easings.net/#easeInElastic

---

### `back()`

```javascript

static back(s)

```

Use with `Animated.parallel()` to create a simple effect where the object animates back slightly as the animation starts.

---

### `bounce()`

```javascript

static bounce(t)

```

Provides a simple bouncing effect.

http://easings.net/#easeInBounce

---

### `bezier()`

```javascript

static bezier(x1, y1, x2, y2)

```

Provides a cubic bezier curve, equivalent to CSS Transitions' `transition-timing-function`.

A useful tool to visualize cubic bezier curves can be found at http://cubic-bezier.com/

---

### `in()`

```javascript
static in easing;
```

Runs an easing function forwards.

---

### `out()`

```javascript

static out(easing)

```

Runs an easing function backwards.

---

### `inOut()`

```javascript

static inOut(easing)

```

Makes any easing function symmetrical. The easing function will run forwards for half of the duration, then backwards for the rest of the duration.
