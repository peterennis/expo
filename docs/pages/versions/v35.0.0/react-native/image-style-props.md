---
id: image-style-props
title: Image Style Props
---

### Props

- [`borderTopRightRadius`](../image-style-props/#bordertoprightradius)
- [`backfaceVisibility`](../image-style-props/#backfacevisibility)
- [`borderBottomLeftRadius`](../image-style-props/#borderbottomleftradius)
- [`borderBottomRightRadius`](../image-style-props/#borderbottomrightradius)
- [`borderColor`](../image-style-props/#bordercolor)
- [`borderRadius`](../image-style-props/#borderradius)
- [`borderTopLeftRadius`](../image-style-props/#bordertopleftradius)
- [`backgroundColor`](../image-style-props/#backgroundcolor)
- [`borderWidth`](../image-style-props/#borderwidth)
- [`opacity`](../image-style-props/#opacity)
- [`overflow`](../image-style-props/#overflow)
- [`resizeMode`](../image-style-props/#resizemode)
- [`tintColor`](../image-style-props/#tintcolor)
- [`overlayColor`](../image-style-props/#overlaycolor)

---

# Reference

## Props

### `borderTopRightRadius`

| Type   | Required |
| ------ | -------- |
| number | No       |

---

### `backfaceVisibility`

| Type                      | Required |
| ------------------------- | -------- |
| enum('visible', 'hidden') | No       |

---

### `borderBottomLeftRadius`

| Type   | Required |
| ------ | -------- |
| number | No       |

---

### `borderBottomRightRadius`

| Type   | Required |
| ------ | -------- |
| number | No       |

---

### `borderColor`

| Type                | Required |
| ------------------- | -------- |
| [color](../colors/) | No       |

---

### `borderRadius`

| Type   | Required |
| ------ | -------- |
| number | No       |

---

### `borderTopLeftRadius`

| Type   | Required |
| ------ | -------- |
| number | No       |

---

### `backgroundColor`

| Type                | Required |
| ------------------- | -------- |
| [color](../colors/) | No       |

---

### `borderWidth`

| Type   | Required |
| ------ | -------- |
| number | No       |

---

### `opacity`

| Type   | Required |
| ------ | -------- |
| number | No       |

---

### `overflow`

| Type                      | Required |
| ------------------------- | -------- |
| enum('visible', 'hidden') | No       |

---

### `resizeMode`

| Type                                                    | Required |
| ------------------------------------------------------- | -------- |
| enum('cover', 'contain', 'stretch', 'repeat', 'center') | No       |

---

### `tintColor`

Changes the color of all the non-transparent pixels to the tintColor.

| Type                | Required |
| ------------------- | -------- |
| [color](../colors/) | No       |

---

### `overlayColor`

When the image has rounded corners, specifying an overlayColor will cause the remaining space in the corners to be filled with a solid color. This is useful in cases which are not supported by the Android implementation of rounded corners:

- Certain resize modes, such as 'contain'
- Animated GIFs

A typical way to use this prop is with images displayed on a solid background and setting the `overlayColor` to the same color as the background.

For details of how this works under the hood, see https://frescolib.org/docs/rounded-corners-and-circles.html

| Type   | Required | Platform |
| ------ | -------- | -------- |
| string | No       | Android  |
