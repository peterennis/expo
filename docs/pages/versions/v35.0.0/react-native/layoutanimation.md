---
id: layoutanimation
title: LayoutAnimation
---

Automatically animates views to their new positions when the next layout happens.

A common way to use this API is to call it before calling `setState`.

Note that in order to get this to work on **Android** you need to set the following flags via `UIManager`:

```java
    UIManager.setLayoutAnimationEnabledExperimental && UIManager.setLayoutAnimationEnabledExperimental(true);
```

### Methods

- [`configureNext`](../layoutanimation/#configurenext)
- [`create`](../layoutanimation/#create)

### Properties

- [`Types`](../layoutanimation/#types)
- [`Properties`](../layoutanimation/#properties)
- [`Presets`](../layoutanimation/#presets)
- [`easeInEaseOut`](../layoutanimation/#easeineaseout)
- [`linear`](../layoutanimation/#linear)
- [`spring`](../layoutanimation/#spring)

---

# Reference

## Methods

### `configureNext()`

```javascript

static configureNext(config, onAnimationDidEnd?)

```

Schedules an animation to happen on the next layout.

#### Parameters:

| Name              | Type     | Required | Description                                                |
| ----------------- | -------- | -------- | ---------------------------------------------------------- |
| config            | object   | Yes      | See config parameters below.                               |
| onAnimationDidEnd | function | No       | Called when the animation finished. Only supported on iOS. |

##### config

- `duration` in milliseconds
- `create`, config for animating in new views (see `Anim` type)
- `update`, config for animating views that have been updated (see `Anim` type)

---

### `create()`

```javascript

static create(duration, type, creationProp)

```

Helper for creating a config for `configureNext`.

## Properties

### Types

An enumerate of animation types to be used in [`create`](../layoutanimation/#create) method.

| Types         |
| ------------- |
| spring        |
| linear        |
| easeInEaseOut |
| easeIn        |
| easeOut       |
| keyboard      |

---

### Properties

An enumerate of object property to be animated, used in [`create`](../layoutanimation/#create) method.

| Properties |
| ---------- |
| opacity    |
| scaleX     |
| scaleY     |
| scaleXY    |

---

### Presets

A set of predefined animation config.

| Presets       | Value                                                                                                                                                                 |
| ------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| easeInEaseOut | `create(300, 'easeInEaseOut', 'opacity')`                                                                                                                             |
| linear        | `create(500, 'linear', 'opacity')`                                                                                                                                    |
| spring        | `{ duration: 700, create: { type: 'linear', property: 'opacity' }, update: { type: 'spring', springDamping: 0.4 }, delete: { type: 'linear', property: 'opacity' } }` |

---

### easeInEaseOut

Shortcut to bind `configureNext()` methods with `Presets.easeInEaseOut`.

---

### linear

Shortcut to bind `configureNext()` methods with `Presets.linear`.

---

### spring

Shortcut to bind `configureNext()` methods with `Presets.spring`.
