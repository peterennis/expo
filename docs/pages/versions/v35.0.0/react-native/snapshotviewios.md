---
id: snapshotviewios
title: SnapshotViewIOS
---

### Props

- [View props...](../view/#props)

* [`onSnapshotReady`](../snapshotviewios/#onsnapshotready)
* [`testIdentifier`](../snapshotviewios/#testidentifier)

---

# Reference

## Props

### `onSnapshotReady`

| Type     | Required |
| -------- | -------- |
| function | No       |

---

### `testIdentifier`

| Type   | Required |
| ------ | -------- |
| string | No       |
