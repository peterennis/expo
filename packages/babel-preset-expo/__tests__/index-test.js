const babel = require('@babel/core');
const fs = require('fs');
const path = require('path');

const preset = require('..');
// Used to distinguish webpack from metro
const WEBPACK_CALLER = { name: 'babel-loader' };

describe.each([['metro'], ['webpack', WEBPACK_CALLER]])('%s', (_name, caller) => {
  it(`compiles sample files`, () => {
    let options = {
      babelrc: false,
      presets: [preset],
      sourceMaps: true,
      caller,
    };
    let samplesPath = path.resolve(__dirname, 'samples');
    let filenames = fs.readdirSync(samplesPath);

    for (let filename of filenames) {
      let { code, map, ast } = babel.transformFileSync(path.join(samplesPath, filename), options);

      expect(code).toBeDefined();
      expect(map).toBeDefined();
      expect(ast).toBeDefined();
    }
  });

  it(`uses the platform's react-native import`, () => {
    let options = {
      babelrc: false,
      presets: [preset],
      filename: 'unknown',
      // Make the snapshot easier to read
      retainLines: true,
      caller,
    };

    let sourceCode = `
import { View } from 'react-native';
`;
    let { code } = babel.transform(sourceCode, options);

    expect(code).toMatchSnapshot();
  });

  it(`aliases @expo/vector-icons`, () => {
    let options = {
      babelrc: false,
      presets: [preset],
      filename: 'unknown',
      // Make the snapshot easier to read
      retainLines: true,
      caller,
    };

    let sourceCode = `
import 'react-native-vector-icons';
require('react-native-vector-icons');
imposter.require('react-native-vector-icons');
imposter.import('react-native-vector-icons');
`;
    let { code } = babel.transform(sourceCode, options);

    expect(code).toMatch(/"@expo\/vector-icons"/);
    expect(code).toMatchSnapshot();
  });

  it(`composes with babel-plugin-module-resolver`, () => {
    let options = {
      babelrc: false,
      presets: [preset],
      plugins: [
        [
          'module-resolver',
          {
            alias: { rn: 'react-native' },
          },
        ],
      ],
      filename: 'unknown',
      // Make the snapshot easier to read
      retainLines: true,
      caller,
    };

    let sourceCode = `
import 'rn';
import 'react-native-vector-icons';
`;
    let { code } = babel.transform(sourceCode, options);

    expect(code).toMatch(/"react-native"/);
    expect(code).toMatch(/"@expo\/vector-icons"/);
    expect(code).toMatchSnapshot();
  });
});

describe('"lazyImports" option', () => {
  it(`defaults to null`, () => {
    let testFilename = path.resolve(__dirname, 'samples', 'Lazy.js');
    let optionsDefault = {
      babelrc: false,
      presets: [preset],
    };
    let { codeDefault } = babel.transformFileSync(testFilename, optionsDefault);

    let optionsNull = {
      babelrc: false,
      presets: [[preset, { lazyImports: null }]],
    };
    let { codeNull } = babel.transformFileSync(testFilename, optionsNull);

    expect(codeDefault).toEqual(codeNull);
  });

  it.each([
    [null],
    [false],
    [true],
    [['inline-comp', './inline-func', '../inline-func-with-side-effects.fx.ts']],
    [name => !(name.endsWith('.fx') || name.endsWith('.fx.js') || name.endsWith('.fx.ts'))],
  ])(`accepts %p`, lazyImportsOption => {
    let testFilename = path.resolve(__dirname, 'samples', 'Lazy.js');
    let options = {
      babelrc: false,
      presets: [[preset, { lazyImports: lazyImportsOption }]],
      // Make the snapshot easier to read
      retainLines: true,
    };

    let { code } = babel.transformFileSync(testFilename, options);
    expect(code).toMatchSnapshot();
  });
});
