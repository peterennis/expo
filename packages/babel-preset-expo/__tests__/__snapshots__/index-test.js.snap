// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"lazyImports" option accepts ["inline-comp", "./inline-func", "../inline-func-with-side-effects.fx.ts"] 1`] = `
"var _interopRequireDefault=require(\\"@babel/runtime/helpers/interopRequireDefault\\");var _interopRequireWildcard=require(\\"@babel/runtime/helpers/interopRequireWildcard\\");Object.defineProperty(exports,\\"__esModule\\",{value:true});exports.default=void 0;var _classCallCheck2=_interopRequireDefault(require(\\"@babel/runtime/helpers/classCallCheck\\"));var _createClass2=_interopRequireDefault(require(\\"@babel/runtime/helpers/createClass\\"));var _possibleConstructorReturn2=_interopRequireDefault(require(\\"@babel/runtime/helpers/possibleConstructorReturn\\"));var _getPrototypeOf2=_interopRequireDefault(require(\\"@babel/runtime/helpers/getPrototypeOf\\"));var _inherits2=_interopRequireDefault(require(\\"@babel/runtime/helpers/inherits\\"));var _expo=require(\\"expo\\");
var _expoAsset=require(\\"expo-asset\\");
var _react=_interopRequireDefault(require(\\"react\\"));
var _reactNative=require(\\"react-native\\");
var _reactRedux=require(\\"react-redux\\");
var _vectorIcons=require(\\"@expo/vector-icons\\");
require(\\"Foo\\");
var Boo=_interopRequireWildcard(require(\\"Boo\\"));
var _fooView=_interopRequireDefault(require(\\"foo-view\\"));
function _inlineComp(){var data=_interopRequireDefault(require(\\"inline-comp\\"));_inlineComp=function _inlineComp(){return data;};return data;}
require(\\"./local-file/i-have-side-effects.fx.js\\");
require(\\"../i-also-have-side-effects.fx\\");
function _inlineFunc(){var data=_interopRequireDefault(require(\\"./inline-func\\"));_inlineFunc=function _inlineFunc(){return data;};return data;}
function _inlineFuncWithSideEffectsFx(){var data=_interopRequireDefault(require(\\"../inline-func-with-side-effects.fx.ts\\"));_inlineFuncWithSideEffectsFx=function _inlineFuncWithSideEffectsFx(){return data;};return data;}var _dec,_class;var


Lazy=(_dec=(0,_reactRedux.connect)(),_dec(_class=function(_React$Component){(0,_inherits2.default)(Lazy,_React$Component);function Lazy(){(0,_classCallCheck2.default)(this,Lazy);return(0,_possibleConstructorReturn2.default)(this,(0,_getPrototypeOf2.default)(Lazy).apply(this,arguments));}(0,_createClass2.default)(Lazy,[{key:\\"componentDidMount\\",value:function componentDidMount()
{
console.log('Calling InlineFuncFromPackage()');
(0,_inlineComp().default)();

console.log('Calling InlineFuncFromFile()');
(0,_inlineFunc().default)();

console.log('Calling Boo.myFunc()');
Boo.myFunc();

_expoAsset.Asset.loadAsync([require(\\"./assets/icon.png\\")]);
}},{key:\\"render\\",value:function render()

{
return(
_react.default.createElement(_reactNative.View,null,
_react.default.createElement(_expo.AppLoading,null),
_react.default.createElement(_vectorIcons.Ionicons,{name:\\"md-options\\",size:28}),
_react.default.createElement(_fooView.default,null)));


}},{key:\\"unusedFunction\\",value:function unusedFunction()

{
(0,_inlineFuncWithSideEffectsFx().default)();
}}]);return Lazy;}(_react.default.Component))||_class);var _default=


Lazy;exports.default=_default;"
`;

exports[`"lazyImports" option accepts [Function anonymous] 1`] = `
"var _interopRequireDefault=require(\\"@babel/runtime/helpers/interopRequireDefault\\");var _interopRequireWildcard=require(\\"@babel/runtime/helpers/interopRequireWildcard\\");Object.defineProperty(exports,\\"__esModule\\",{value:true});exports.default=void 0;function _classCallCheck2(){var data=_interopRequireDefault(require(\\"@babel/runtime/helpers/classCallCheck\\"));_classCallCheck2=function _classCallCheck2(){return data;};return data;}function _createClass2(){var data=_interopRequireDefault(require(\\"@babel/runtime/helpers/createClass\\"));_createClass2=function _createClass2(){return data;};return data;}function _possibleConstructorReturn2(){var data=_interopRequireDefault(require(\\"@babel/runtime/helpers/possibleConstructorReturn\\"));_possibleConstructorReturn2=function _possibleConstructorReturn2(){return data;};return data;}function _getPrototypeOf2(){var data=_interopRequireDefault(require(\\"@babel/runtime/helpers/getPrototypeOf\\"));_getPrototypeOf2=function _getPrototypeOf2(){return data;};return data;}function _inherits2(){var data=_interopRequireDefault(require(\\"@babel/runtime/helpers/inherits\\"));_inherits2=function _inherits2(){return data;};return data;}function _expo(){var data=require(\\"expo\\");_expo=function _expo(){return data;};return data;}
function _expoAsset(){var data=require(\\"expo-asset\\");_expoAsset=function _expoAsset(){return data;};return data;}
function _react(){var data=_interopRequireDefault(require(\\"react\\"));_react=function _react(){return data;};return data;}
function _reactNative(){var data=require(\\"react-native\\");_reactNative=function _reactNative(){return data;};return data;}
function _reactRedux(){var data=require(\\"react-redux\\");_reactRedux=function _reactRedux(){return data;};return data;}
function _vectorIcons(){var data=require(\\"@expo/vector-icons\\");_vectorIcons=function _vectorIcons(){return data;};return data;}
require(\\"Foo\\");
function Boo(){var data=_interopRequireWildcard(require(\\"Boo\\"));Boo=function Boo(){return data;};return data;}
function _fooView(){var data=_interopRequireDefault(require(\\"foo-view\\"));_fooView=function _fooView(){return data;};return data;}
function _inlineComp(){var data=_interopRequireDefault(require(\\"inline-comp\\"));_inlineComp=function _inlineComp(){return data;};return data;}
require(\\"./local-file/i-have-side-effects.fx.js\\");
require(\\"../i-also-have-side-effects.fx\\");
function _inlineFunc(){var data=_interopRequireDefault(require(\\"./inline-func\\"));_inlineFunc=function _inlineFunc(){return data;};return data;}
var _inlineFuncWithSideEffectsFx=_interopRequireDefault(require(\\"../inline-func-with-side-effects.fx.ts\\"));var _dec,_class;var


Lazy=(_dec=(0,_reactRedux().connect)(),_dec(_class=function(_React$Component){(0,_inherits2().default)(Lazy,_React$Component);function Lazy(){(0,_classCallCheck2().default)(this,Lazy);return(0,_possibleConstructorReturn2().default)(this,(0,_getPrototypeOf2().default)(Lazy).apply(this,arguments));}(0,_createClass2().default)(Lazy,[{key:\\"componentDidMount\\",value:function componentDidMount()
{
console.log('Calling InlineFuncFromPackage()');
(0,_inlineComp().default)();

console.log('Calling InlineFuncFromFile()');
(0,_inlineFunc().default)();

console.log('Calling Boo.myFunc()');
Boo().myFunc();

_expoAsset().Asset.loadAsync([require(\\"./assets/icon.png\\")]);
}},{key:\\"render\\",value:function render()

{
return(
_react().default.createElement(_reactNative().View,null,
_react().default.createElement(_expo().AppLoading,null),
_react().default.createElement(_vectorIcons().Ionicons,{name:\\"md-options\\",size:28}),
_react().default.createElement(_fooView().default,null)));


}},{key:\\"unusedFunction\\",value:function unusedFunction()

{
(0,_inlineFuncWithSideEffectsFx.default)();
}}]);return Lazy;}(_react().default.Component))||_class);var _default=


Lazy;exports.default=_default;"
`;

exports[`"lazyImports" option accepts false 1`] = `
"var _interopRequireDefault=require(\\"@babel/runtime/helpers/interopRequireDefault\\");var _interopRequireWildcard=require(\\"@babel/runtime/helpers/interopRequireWildcard\\");Object.defineProperty(exports,\\"__esModule\\",{value:true});exports.default=void 0;var _classCallCheck2=_interopRequireDefault(require(\\"@babel/runtime/helpers/classCallCheck\\"));var _createClass2=_interopRequireDefault(require(\\"@babel/runtime/helpers/createClass\\"));var _possibleConstructorReturn2=_interopRequireDefault(require(\\"@babel/runtime/helpers/possibleConstructorReturn\\"));var _getPrototypeOf2=_interopRequireDefault(require(\\"@babel/runtime/helpers/getPrototypeOf\\"));var _inherits2=_interopRequireDefault(require(\\"@babel/runtime/helpers/inherits\\"));var _expo=require(\\"expo\\");
var _expoAsset=require(\\"expo-asset\\");
var _react=_interopRequireDefault(require(\\"react\\"));
var _reactNative=require(\\"react-native\\");
var _reactRedux=require(\\"react-redux\\");
var _vectorIcons=require(\\"@expo/vector-icons\\");
require(\\"Foo\\");
var Boo=_interopRequireWildcard(require(\\"Boo\\"));
var _fooView=_interopRequireDefault(require(\\"foo-view\\"));
var _inlineComp=_interopRequireDefault(require(\\"inline-comp\\"));
require(\\"./local-file/i-have-side-effects.fx.js\\");
require(\\"../i-also-have-side-effects.fx\\");
var _inlineFunc=_interopRequireDefault(require(\\"./inline-func\\"));
var _inlineFuncWithSideEffectsFx=_interopRequireDefault(require(\\"../inline-func-with-side-effects.fx.ts\\"));var _dec,_class;var


Lazy=(_dec=(0,_reactRedux.connect)(),_dec(_class=function(_React$Component){(0,_inherits2.default)(Lazy,_React$Component);function Lazy(){(0,_classCallCheck2.default)(this,Lazy);return(0,_possibleConstructorReturn2.default)(this,(0,_getPrototypeOf2.default)(Lazy).apply(this,arguments));}(0,_createClass2.default)(Lazy,[{key:\\"componentDidMount\\",value:function componentDidMount()
{
console.log('Calling InlineFuncFromPackage()');
(0,_inlineComp.default)();

console.log('Calling InlineFuncFromFile()');
(0,_inlineFunc.default)();

console.log('Calling Boo.myFunc()');
Boo.myFunc();

_expoAsset.Asset.loadAsync([require(\\"./assets/icon.png\\")]);
}},{key:\\"render\\",value:function render()

{
return(
_react.default.createElement(_reactNative.View,null,
_react.default.createElement(_expo.AppLoading,null),
_react.default.createElement(_vectorIcons.Ionicons,{name:\\"md-options\\",size:28}),
_react.default.createElement(_fooView.default,null)));


}},{key:\\"unusedFunction\\",value:function unusedFunction()

{
(0,_inlineFuncWithSideEffectsFx.default)();
}}]);return Lazy;}(_react.default.Component))||_class);var _default=


Lazy;exports.default=_default;"
`;

exports[`"lazyImports" option accepts null 1`] = `
"var _interopRequireDefault=require(\\"@babel/runtime/helpers/interopRequireDefault\\");var _interopRequireWildcard=require(\\"@babel/runtime/helpers/interopRequireWildcard\\");Object.defineProperty(exports,\\"__esModule\\",{value:true});exports.default=void 0;var _classCallCheck2=_interopRequireDefault(require(\\"@babel/runtime/helpers/classCallCheck\\"));var _createClass2=_interopRequireDefault(require(\\"@babel/runtime/helpers/createClass\\"));var _possibleConstructorReturn2=_interopRequireDefault(require(\\"@babel/runtime/helpers/possibleConstructorReturn\\"));var _getPrototypeOf2=_interopRequireDefault(require(\\"@babel/runtime/helpers/getPrototypeOf\\"));var _inherits2=_interopRequireDefault(require(\\"@babel/runtime/helpers/inherits\\"));var _expo=require(\\"expo\\");
var _expoAsset=require(\\"expo-asset\\");
var _react=_interopRequireDefault(require(\\"react\\"));
var _reactNative=require(\\"react-native\\");
var _reactRedux=require(\\"react-redux\\");
var _vectorIcons=require(\\"@expo/vector-icons\\");
require(\\"Foo\\");
var Boo=_interopRequireWildcard(require(\\"Boo\\"));
var _fooView=_interopRequireDefault(require(\\"foo-view\\"));
var _inlineComp=_interopRequireDefault(require(\\"inline-comp\\"));
require(\\"./local-file/i-have-side-effects.fx.js\\");
require(\\"../i-also-have-side-effects.fx\\");
var _inlineFunc=_interopRequireDefault(require(\\"./inline-func\\"));
var _inlineFuncWithSideEffectsFx=_interopRequireDefault(require(\\"../inline-func-with-side-effects.fx.ts\\"));var _dec,_class;var


Lazy=(_dec=(0,_reactRedux.connect)(),_dec(_class=function(_React$Component){(0,_inherits2.default)(Lazy,_React$Component);function Lazy(){(0,_classCallCheck2.default)(this,Lazy);return(0,_possibleConstructorReturn2.default)(this,(0,_getPrototypeOf2.default)(Lazy).apply(this,arguments));}(0,_createClass2.default)(Lazy,[{key:\\"componentDidMount\\",value:function componentDidMount()
{
console.log('Calling InlineFuncFromPackage()');
(0,_inlineComp.default)();

console.log('Calling InlineFuncFromFile()');
(0,_inlineFunc.default)();

console.log('Calling Boo.myFunc()');
Boo.myFunc();

_expoAsset.Asset.loadAsync([require(\\"./assets/icon.png\\")]);
}},{key:\\"render\\",value:function render()

{
return(
_react.default.createElement(_reactNative.View,null,
_react.default.createElement(_expo.AppLoading,null),
_react.default.createElement(_vectorIcons.Ionicons,{name:\\"md-options\\",size:28}),
_react.default.createElement(_fooView.default,null)));


}},{key:\\"unusedFunction\\",value:function unusedFunction()

{
(0,_inlineFuncWithSideEffectsFx.default)();
}}]);return Lazy;}(_react.default.Component))||_class);var _default=


Lazy;exports.default=_default;"
`;

exports[`"lazyImports" option accepts true 1`] = `
"var _interopRequireDefault=require(\\"@babel/runtime/helpers/interopRequireDefault\\");var _interopRequireWildcard=require(\\"@babel/runtime/helpers/interopRequireWildcard\\");Object.defineProperty(exports,\\"__esModule\\",{value:true});exports.default=void 0;function _classCallCheck2(){var data=_interopRequireDefault(require(\\"@babel/runtime/helpers/classCallCheck\\"));_classCallCheck2=function _classCallCheck2(){return data;};return data;}function _createClass2(){var data=_interopRequireDefault(require(\\"@babel/runtime/helpers/createClass\\"));_createClass2=function _createClass2(){return data;};return data;}function _possibleConstructorReturn2(){var data=_interopRequireDefault(require(\\"@babel/runtime/helpers/possibleConstructorReturn\\"));_possibleConstructorReturn2=function _possibleConstructorReturn2(){return data;};return data;}function _getPrototypeOf2(){var data=_interopRequireDefault(require(\\"@babel/runtime/helpers/getPrototypeOf\\"));_getPrototypeOf2=function _getPrototypeOf2(){return data;};return data;}function _inherits2(){var data=_interopRequireDefault(require(\\"@babel/runtime/helpers/inherits\\"));_inherits2=function _inherits2(){return data;};return data;}var _expo=require(\\"expo\\");
var _expoAsset=require(\\"expo-asset\\");
function _react(){var data=_interopRequireDefault(require(\\"react\\"));_react=function _react(){return data;};return data;}
function _reactNative(){var data=require(\\"react-native\\");_reactNative=function _reactNative(){return data;};return data;}
function _reactRedux(){var data=require(\\"react-redux\\");_reactRedux=function _reactRedux(){return data;};return data;}
function _vectorIcons(){var data=require(\\"@expo/vector-icons\\");_vectorIcons=function _vectorIcons(){return data;};return data;}
require(\\"Foo\\");
function Boo(){var data=_interopRequireWildcard(require(\\"Boo\\"));Boo=function Boo(){return data;};return data;}
function _fooView(){var data=_interopRequireDefault(require(\\"foo-view\\"));_fooView=function _fooView(){return data;};return data;}
function _inlineComp(){var data=_interopRequireDefault(require(\\"inline-comp\\"));_inlineComp=function _inlineComp(){return data;};return data;}
require(\\"./local-file/i-have-side-effects.fx.js\\");
require(\\"../i-also-have-side-effects.fx\\");
var _inlineFunc=_interopRequireDefault(require(\\"./inline-func\\"));
var _inlineFuncWithSideEffectsFx=_interopRequireDefault(require(\\"../inline-func-with-side-effects.fx.ts\\"));var _dec,_class;var


Lazy=(_dec=(0,_reactRedux().connect)(),_dec(_class=function(_React$Component){(0,_inherits2().default)(Lazy,_React$Component);function Lazy(){(0,_classCallCheck2().default)(this,Lazy);return(0,_possibleConstructorReturn2().default)(this,(0,_getPrototypeOf2().default)(Lazy).apply(this,arguments));}(0,_createClass2().default)(Lazy,[{key:\\"componentDidMount\\",value:function componentDidMount()
{
console.log('Calling InlineFuncFromPackage()');
(0,_inlineComp().default)();

console.log('Calling InlineFuncFromFile()');
(0,_inlineFunc.default)();

console.log('Calling Boo.myFunc()');
Boo().myFunc();

_expoAsset.Asset.loadAsync([require(\\"./assets/icon.png\\")]);
}},{key:\\"render\\",value:function render()

{
return(
_react().default.createElement(_reactNative().View,null,
_react().default.createElement(_expo.AppLoading,null),
_react().default.createElement(_vectorIcons().Ionicons,{name:\\"md-options\\",size:28}),
_react().default.createElement(_fooView().default,null)));


}},{key:\\"unusedFunction\\",value:function unusedFunction()

{
(0,_inlineFuncWithSideEffectsFx.default)();
}}]);return Lazy;}(_react().default.Component))||_class);var _default=


Lazy;exports.default=_default;"
`;

exports[`metro aliases @expo/vector-icons 1`] = `
"
require(\\"@expo/vector-icons\\");
require(\\"@expo/vector-icons\\");
imposter.require('react-native-vector-icons');
imposter.import('react-native-vector-icons');"
`;

exports[`metro composes with babel-plugin-module-resolver 1`] = `
"
require(\\"react-native\\");
require(\\"@expo/vector-icons\\");"
`;

exports[`metro uses the platform's react-native import 1`] = `
"
var _reactNative=require(\\"react-native\\");"
`;

exports[`webpack aliases @expo/vector-icons 1`] = `
"
import\\"@expo/vector-icons\\";
require(\\"@expo/vector-icons\\");
imposter.require('react-native-vector-icons');
imposter.import('react-native-vector-icons');"
`;

exports[`webpack composes with babel-plugin-module-resolver 1`] = `
"
import\\"react-native\\";
import\\"@expo/vector-icons\\";"
`;

exports[`webpack uses the platform's react-native import 1`] = `"import View from\\"react-native-web/dist/exports/View\\";"`;
